src/components/carousel/index.js

```diff
@@ -114,49 +114,75 @@ export default class Carousel extends BaseComponent {
   constructor(props) {
     super(props);
 
-    const themeProps = this.getThemeProps();
     this.carousel = React.createRef();
-    const defaultPageWidth = (themeProps.loop || !themeProps.pageWidth) ? Constants.screenWidth : themeProps.pageWidth;
-    
+
+    const defaultPageWidth = (props.loop || !props.pageWidth) ? Constants.screenWidth : props.pageWidth;
+
     this.state = {
       containerWidth: undefined,
-      currentPage: this.shouldUsePageWidth() ? this.getCalcIndex(themeProps.initialPage) : themeProps.initialPage,
-      currentStandingPage: themeProps.initialPage,
+      currentPage: this.shouldUsePageWidth() ? this.getCalcIndex(props.initialPage) : props.initialPage,
+      currentStandingPage: props.initialPage,
       pageWidth: defaultPageWidth,
-      initialOffset: {x: presenter.calcOffset(themeProps, {currentPage: themeProps.initialPage, pageWidth: defaultPageWidth})}
+      initialOffset: {
+        x: presenter.calcOffset(props, {
+          currentPage: props.initialPage,
+          pageWidth: defaultPageWidth
+        })
+      },
+      prevProps: props
     };
   }
 
   componentDidMount() {
     Constants.addDimensionsEventListener(this.onOrientationChanged);
 
-    if (this.getThemeProps().autoplay) {
+    if (this.props.autoplay) {
       this.startAutoPlay();
     }
   }
 
-  // TODO: change to getDerivedStateFromProps (requires changing from BaseComponent to asBaseComponent)
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const {currentPage} = this.state;
-    const {pageWidth: nexPageWidth} = nextProps;
-    const {pageWidth} = this.getThemeProps();
+  componentWillUnmount() {
+    Constants.removeDimensionsEventListener(this.onOrientationChanged);
 
-    if (pageWidth !== nexPageWidth) {
-      const pageWidth = nexPageWidth;
-      this.setState({
+    clearInterval(this.autoplayTimer);
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const {currentPage, prevProps} = prevState;
```

Comment:
Is there a reason to keep the whole "prevProps" in state? 
Why not just keep the initialPageWidth in state and check against its value.

Reply:
I'm using 'pageWidth', 'containerMarginHorizontal', 'loop' for the prevProps and passing them to 'presenter.calcOffset()'

Reply:
👍